packaging/tags.py

```diff
@@ -733,9 +732,96 @@ def _manylinux_tags(linux, arch):
+    platform-specific built distributions. Typically includes the OS name and
+    version and the architecture (as supplied by 'os.uname()'), although the
+    exact information included depends on the OS; eg. on Linux, the kernel
+    version isn't particularly important.
+
+    Examples of returned values:
+       linux-i586
+       linux-alpha (?)
+       solaris-2.6-sun4u
+
+    Windows will return one of:
+       win-amd64 (64bit Windows on AMD64 (aka x86_64, Intel64, EM64T, etc)
+       win32 (all others - specifically, sys.platform is returned)
+
+    For other non-POSIX platforms, currently just returns 'sys.platform'.
+    """
+    if os.name == "nt":
+        if "amd64" in sys.version.lower():
+            return "win-amd64"
+        if "(arm)" in sys.version.lower():
+            return "win-arm32"
+        if "(arm64)" in sys.version.lower():
+            return "win-arm64"
+        return sys.platform
+
+    # Set for cross builds explicitly
+    if "_PYTHON_HOST_PLATFORM" in os.environ:
+        return os.environ["_PYTHON_HOST_PLATFORM"]
+
+    if os.name != "posix" or not hasattr(os, "uname"):
+        # XXX what about the architecture? NT is Intel or Alpha,
+        # Mac OS is M68k or PPC, etc.
+        return sys.platform
+
+    # Try to distinguish various flavours of Unix
+
+    (osname, host, release, version, machine) = os.uname()
+
+    # Convert the OS name to lowercase, remove '/' characters, and translate
+    # spaces (for "Power Macintosh")
+    osname = osname.lower().replace("/", "")
+    machine = machine.replace(" ", "_")
+    machine = machine.replace("/", "-")
+
+    if osname[:5] == "linux":
+        # At least on Linux/Intel, 'machine' is the processor -- i386, etc.
+        # XXX what about Alpha, SPARC, etc?
+        return "{}-{}".format(osname, machine)
+    elif osname[:5] == "sunos":
+        if release[0] >= "5":  # SunOS 5 == Solaris 2
+            osname = "solaris"
+            release = "{}.{}".format(int(release[0]) - 3, release[2:])
+            # We can't use "platform.architecture()[0]" because a
+            # bootstrap problem. We use a dict to get an error
+            # if some suspicious happens.
+            bitness = {2147483647: "32bit", 9223372036854775807: "64bit"}
+            machine += ".{}".format(bitness[sys.maxsize])
+        # fall through to standard osname-release-machine representation
+    elif osname[:3] == "aix":
+        from _aix_support import aix_platform
+
+        return cast(str, aix_platform())
+    elif osname[:6] == "cygwin":
+        osname = "cygwin"
+        if PY2:  # Python 2 does not have re.ASCII.
+            rel_re = re.compile(r"[\d.]+")
+        else:
+            rel_re = re.compile(r"[\d.]+", re.ASCII)
+        m = rel_re.match(release)
+        if m:
+            release = m.group()
+    elif osname[:6] == "darwin":
+        import sysconfig
+        import _osx_support
+
+        osname, release, machine = _osx_support.get_platform_osx(
```

Comment:
My only query is whether this should rely on `_osx_support` (which I assume is a core CPython module?) or reproduce whatever it's doing.

Reply:
Correct, it’s a helper module in stdlib. The function is quite complicated and I’d very much keep referencing the stdlib if possible. Not sure who to ask about whether this is a good idea.

Reply:
@ronaldoussoren @ned-deily

Reply:
Looks like the equivalent public API for this is `sysconfig.get_platform()`, so it should be an easy swap. It might even make a suitable fallback for all cases after you've gotten past the initial tests. I suspect you'll still want your own version of it for backporting fixes/changes, but if the stdlib functionality is fine for this bit then you may as well use as much as you can/like.

Reply:
That said, I only compared to the version of sysconfig in master... it might be that there are already improvements that aren't in 3.9, and so you'll still need the logic you've got here. But the `_osx_support` reference can't really be any worse in earlier versions.

Reply:
`git blame` the everything that changed in that function during the recent 5 years are about Windows (ARM support), AIX improvements, and dropping BSD/OS support, so it’s probably safe. I’ll switch that API.

Reply:
Actually, the same also applies to `_aix_support` (prior to 3.9 it’s just `{osname}-{release}-{machine}`, and was expanded to support PEP 425), so I replaced that as well.